tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,218 @@
+			Skip("Skipping the deployment of the operator")
+		}
+
+		When("installed via helm install", func() {
+			BeforeAll(func() {
+				Expect(kubectl.CreateNamespace(namespace)).To(Succeed(), "Namespace failed to be created")
+
+				extraArg := ""
+				if ocp {
+					extraArg = "--set=platform=openshift"
+				}
+
+				Expect(helm.Install("sail-operator", filepath.Join(baseDir, "chart"), "--namespace "+namespace, "--set=image="+image, extraArg)).
+					Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable), "Operator deployment is not Available; unexpected Condition")
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=sail-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("sail-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(crds), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed(), "Namespace failed to be created; unexpected error")
+				})
+
+				When("the resource is created", func() {
+					BeforeAll(func() {
+						istioYAML := `
+apiVersion: operator.istio.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: %s`
+						istioYAML = fmt.Sprintf(istioYAML, version, controlPlaneNamespace)
+						fmt.Printf("Istio CR YAML: %s\n", istioYAML)
+						Expect(kubectl.ApplyString(istioYAML)).
+							Should(Succeed(), "Istio CR failed to be created; unexpected error")
+						Success("Istio CR created")
+					})
+
+					It("updates the Istio resource status to Reconcilied and Ready", func() {
+						Eventually(kubectl.GetCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReconcilied), "Istio is not Reconcilied; unexpected Condition")
+
+						Eventually(kubectl.GetCondition).
+							WithArguments(controlPlaneNamespace, "istio", istioName).
+							Should(ContainElement(resourceReady), "Istio is not Ready; unexpected Condition")
+
+						Success("Istio resource is Reconcilied and Ready")
+					})
+
+					It("deploys istiod", func() {
+						Expect(kubectl.GetDeploymentNames(controlPlaneNamespace)).
+							To(Equal([]string{"istiod"}), "Istiod deployment is not present; expected list to be equal")
```

Comment:
Not sure what "expected list to be equal" means from the perspective of someone running this test. It's also not really necessary, as ginkgo will print out expected `<contents of the list> to equal []string{"istiod"}`. What the failure message should explain what this list is, since this is not clear from the output. So, it should say something like "unexpected list of deployments". 

Please review the other failure messages if they provide any missing information. If not, we should remove them (or remove just the second part of each message).